### ISSUE TITLE
[$20][MSFT-59] [Registration] - The color contrast is very low for Join Button.

### ISSUE THREAD

**Issue by veshu** [topcoder-platform/community-app#2775]
**Steps to Reproduce**
1. Go to https://accounts.topcoder.com/member/registration
2. Inspect Element
3. Observe the Join Button color. 

**Expected Result**
Expected contrast ratio of 4.5:1. 

**Actual Result**
Element has insufficient color contrast of 1.5.1 (foreground color: #FFFFFF, background color: ##d1d3d4)

**Device**: Desktop/Labtop
**Operating System**: macOS
**Browser**: Chrome

**WCAG Category**: 1. Perceivable
**WCAG Level**: AA
**WCAG Success Criterion**: 1.4.3 Contrast (minimum)
**Screen Reader Used**: 
**Accessibility Audit Tool used**: Other

**Attachment**
https://topcodermsft-my.sharepoint.com/personal/pd-topcoder_topcodermsft_onmicrosoft_com/Documents/Forms/All.aspx?cid=ae14bf9a%2D590b%2D4bb4%2Da40a%2D4c6822c880a4&FolderCTID=0x0120005C598A51049FC14CBA882E1AEE168F51&id=%2Fpersonal%2Fpd%2Dtopcoder%5Ftopcodermsft%5Fonmicrosoft%5Fcom%2FDocuments%2FMSFT%2DTeams%2DQA%2FJune%202019%2F29%20Jun%2FTest%20Results%2FBug%20Videos%20%26%20Log%20Files%2FCreeya%2FBug%2059

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096331 has been created for this ticket.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096331 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096331 has been updated - it has been assigned to PkDurlabhji.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@codeMinter PR https://github.com/appirio-tech/accounts-app/pull/242

**Comment by Dara-K** [comment]
That button is disabled now and will become enabled once the user fills all the fields.

**Comment by tosha5252** [comment]
(disabled) color changed to #C3C1C1
Contrast Ratio: 1.79:1...does not meet WCAG requirements

**Comment by PrakashDurlabhji** [comment]
@veshu @tosha5252 why is PR reverted for it?
https://github.com/appirio-tech/accounts-app/pull/259
confused

**Comment by veshu** [comment]
@mishacucicea FYI

**Comment by mishacucicea** [comment]
@PrakashDurlabhji Please follow the issues/conversations. 
If an issue it's QA Fail or tcx_Feedback it needs to be addressed.
By default a 'QA Fail' needs to be reverted, but if you answer fast it won't be.
Please do the adjustments and create another pull request, it's a simple change.

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30097377 has been created for this ticket.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30097377 has been updated - it has been assigned to PkDurlabhji.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@mishacucicea PR raised again https://github.com/appirio-tech/accounts-app/pull/267

**Comment by mishacucicea** [comment]
@PrakashDurlabhji  You introduced a color that's not needed and the contrast was still not fixed.
@Dara-K While fixing the colors, I've noticed that disabled buttons have an opacity which will make them less visible, even if the Tool says that contrast is OK. Not sure exactly if we should used other errors.
![image](https://user-images.githubusercontent.com/1721199/62037908-60c6e300-b1fd-11e9-857b-e7a1305912ae.png)

**Comment by mishacucicea** [comment]
@PrakashDurlabhji Please do not pick-up new items to work on if you have pending items already, if needed I will enforce that.

**Comment by tosha5252** [comment]
The color has been changed to #5d5d66, but would also like to know from @Dara-K if opacity level needs to be adjusted or removed...@mishacucicea, would another issue be created to change the opacity level, or just use this one?

**Comment by mishacucicea** [comment]
@tosha5252 - We'll use a new one if needed, to keep it simple.

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30097377 has been updated - it has been assigned to PkDurlabhji.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by Dara-K** [comment]
For this one, let's change the color of the Join button and Show button when disabled to #6B6B6B. No need to use transparency for them as that will lower the contrast only and will make it difficult to perceive for the users.

**Comment by PrakashDurlabhji** [comment]
@mishacucicea sorry for delay  but all will be tested now and perfect, wont take time now. I am raising PR now which satisfies above comment of Dara-K, I also saw your commits on my PR which is merged now, thanks for that.

**Comment by PrakashDurlabhji** [comment]
@mishacucicea @tosha5252 new PR and locally tested perfectly. https://github.com/appirio-tech/accounts-app/pull/278

**Comment by mishacucicea** [comment]
@PrakashDurlabhji 
 - You've introduced an unneeded css class
 - Raised complexity because you missed the `opacity` keyword in the discussion above and the Join didn't had enough contrast in grey because of the default opacity of 0.3.

I have fix it, take a look at the PR, hope you learn from it.

**Comment by PrakashDurlabhji** [comment]
@mishacucicea thanks for help but you said we will handle opacity in another ticket so I did not touch it.

**Comment by drasticdpk** [comment]
The color code suggested by @Dara-K has not been updated. 
![image](https://user-images.githubusercontent.com/25297859/62129109-9cdd6f00-b2f5-11e9-8265-19ac035529d7.png)

**Comment by drasticdpk** [comment]
Hi @PrakashDurlabhji , 
Below is my finding , pease suggest if i was wrong. It would be great if you share your finding screenshot too. 
https://monosnap.com/file/xn2B7DuEzjALghY11ju9KgsS2CtEHZ
https://monosnap.com/file/fzqalFkSaMBriTYYdFqW1lfNxubcbq

**Comment by drasticdpk** [comment]
Now verified on google chrome.

**Comment by drasticdpk** [comment]
@PrakashDurlabhji @Oanh-and-only-Oanh  it was pass on test server. Verified on Google chrome.

**Comment by nithyaasworld** [comment]
Its failing in the following places:

1. Show button in disabled state
2. Join button in the following scenario: 
      a: Just enter "a" in the country field and go to any other field
      b: It shows a message asking the user to choose country from the list
      c: Now enter another "a" and choose a country from the list
      d: Check the Join button disabled state not being in WCAG Standards

![image](https://user-images.githubusercontent.com/5114190/62189974-ec6f7980-b38d-11e9-9e52-148d541b68a8.png)

![image](https://user-images.githubusercontent.com/5114190/62189997-f6917800-b38d-11e9-8bfd-e82ea1c89aa9.png)

**Comment by nithyaasworld** [comment]
@PrakashDurlabhji Please check above. Thanks.

**Comment by nithyaasworld** [comment]
@PrakashDurlabhji, Please wait.

CC: @nkumar-topcoder

**Comment by nkumar-topcoder** [comment]
@PrakashDurlabhji @mishacucicea can you pls look into above comment, fix  and test locally. 
### Note: Very important: Ensure that you are talking latest files for this issue fix  from **DEV** only. Else, merging will be a headache.

**Comment by nkumar-topcoder** [comment]
@PrakashDurlabhji @mishacucicea  #2868, #2775 and #2774 all fix overlap with similar files, earlier merges failed. Can you please look into The PR are 248, 278.

**Comment by PrakashDurlabhji** [comment]
@nithyaasworld 

1. I went through your comment https://github.com/topcoder-platform/community-app/issues/2775#issuecomment-516722407

and for the same additional comment has been added at https://github.com/appirio-tech/accounts-app/pull/278/commits/1875bfd0134534eff7da91eb5fda593483f2f4f8

@mishacucicea and me will sync it to make it work and satisfy requirements.

2. Regarding $tc-white-cream, i can see that variable is present in "qa-accessibility" branch but not in "dev" branch. hence it is saying $tc-white-cream not present

and I guess it is removed by someone while resolving and merging conflicts.
lets keep that aside and lets complete this immediately.

I may reply a bit late but kindly do not assign all my 3 issues to someone else, I will complete ASAP.

Note: I am just confused regardign taking latest from dev? means I need to make a new PR which I am aware:

1. so I need to take latest from dev branch? 
2. and make code changes, but raise PR against "qa-accessiblity"?
3. or I need to make changes, and raise PR against "dev" branch itself?

kindly confirm on above.

@veshu

**Comment by nkumar-topcoder** [comment]
@PrakashDurlabhji The reason to take latest from DEV branch is that there could be other `fixes` which might have been already push to DEV branch for `the file` you are about to fix. This causes merge issue later. If there is conflict then it's fine.

**Comment by PrakashDurlabhji** [comment]
@nkumar-topcoder yes got some idea, trying it. Though it is difficult.

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30097377 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by lakshmiathreya** [comment]
<img width="1228" alt="Screenshot 2019-08-05 at 11 24 21 AM" src="https://user-images.githubusercontent.com/51733042/62441904-9c205f00-b773-11e9-9572-84442f189611.png">
<img width="1258" alt="Screenshot 2019-08-05 at 11 24 06 AM" src="https://user-images.githubusercontent.com/51733042/62441913-a3476d00-b773-11e9-9a7c-0ab5508168f0.png">
Please see the screenshots attached - The disabled Join button has different colors when Country is selected and it is yet to be selected. Disabled Join button should have consistent color right - pls clarify @Dara-K. fyi @nithyaasworld @tosha5252 @Dara-K @nkumar-topcoder

**Comment by PrakashDurlabhji** [comment]
@veshu apology for delay,I am making PR for this and other issues connected with this.

**Comment by lakshmiathreya** [comment]
Partially filled password also has the same issue. Pls see the screenshot.
<img width="1369" alt="Screenshot 2019-08-05 at 11 51 53 AM" src="https://user-images.githubusercontent.com/51733042/62443087-90369c00-b777-11e9-862e-0c124b8b3999.png">

**Comment by tosha5252** [comment]
![country](https://user-images.githubusercontent.com/6400904/62505944-3d370600-b7c4-11e9-9326-c25b0cf20fbc.png)

Witnessing the same as @lakshmiathreya...when the country is picked, button changes

**Comment by PrakashDurlabhji** [comment]
@tosha5252 @lakshmiathreya yes because buttons are different in both cases and code was removed hence changes not visible.

**Comment by tosha5252** [comment]
![country2](https://user-images.githubusercontent.com/6400904/62506351-e6323080-b7c5-11e9-8ce5-99ea6638741c.png)
check the code....twice for the same button

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30097377 has been updated - it has been assigned to rohitgupta_.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by mishacucicea** [comment]
@PrakashDurlabhji , I've assigned to @r0hit-gupta as we need a fast fix.
Please contact me on Slack to discuss about compensating for part of your work on this item.
If you are not on Topcoder Community Slack, please register using [this link](https://topcoder.us13.list-manage.com/track/click?u=65bd5a1857b73643aad556093&id=84b18bc8fd&e=af22e6fdea).

**Comment by r0hit-gupta** [comment]
@mishacucicea @nkumar-topcoder PR at https://github.com/appirio-tech/accounts-app/pull/294 
I have also fixed the Terms & Privacy Policy not highlighting on tab focus. Please review.

![image](https://user-images.githubusercontent.com/16841686/62534468-e1718900-b866-11e9-9fe7-4aded65a3b7a.png)

**Comment by nkumar-topcoder** [comment]
@lakshmiathreya @drasticdpk this is re-fixed and deployed to `qa-accessbility` branch. In case if you find the issue is not fixed let me know i will deploy to `DEV` branch (as we found few had few cases like this earlier)

**Comment by tosha5252** [comment]
The button color is correct and it doesn't change when entering country...

Windows 10 env
Browser: Chrome

**Comment by tosha5252** [comment]
The button color is correct: #5d5d66! 

Windows 10 env
Browser: Chrome

**Comment by tosha5252** [comment]
The button color is correct: #5d5d66

Windows 10 env
Browser: Chrome

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30097377<br/><br/>```This is an automated message for crazyk via Topcoder X```
